### DISCUSSION TITLE
Why is SDL2 built manually using configure?

### DISCUSSION THREAD

**Discussion by pypy702** [Homebrew/discussions#882] [Everyday usage]
Here - https://wiki.libsdl.org/Installation#Mac_OS_X - they specify that the recommended way to build sdl2 on Mac OS is to use the gcc-fat script. However, here it is built manuially using configure.
https://github.com/Homebrew/homebrew-core/blob/6b39b4ba5d1957fe13e877014e468d73c77cc820/Formula/sdl2.rb#L41-L43

Because you use configure manually, it targets the platform that is being built on.. However, if you use the script, it targets macOS 10.6 (useful if you are bundling libraries).

**Comment by carlocab** [comment]
We actually do want to target the current platform, so sticking to `configure` seems like the way to go.

**Reply by pypy702** [reply]
Then the resulting .app wouldn't be runnable on versions less than the version the program was built on.
I want to be able to distribute the resulting program to users, and some may be on macOS versions less than me.

**Reply by jonchang** [reply]
That is not a supported use case for formulae in Homebrew/core. You're better off doing this as a fork in your own tap, using a different package manager, or writing your own scripts.

**Reply by pypy702** [reply]
But I see no harm in changing to the recommended script if it would bring benefits.

**Reply by jonchang** [reply]
We can't guarantee that software built for one version of macOS will run on earlier or later versions as our continuous integration doesn't test for this. We do not have the resources to expand our testing to cover this use case. 

Packaging software for correct redistribution in downstream .apps for older versions of macOS is not something supported in Homebrew/core. You'd be better served doing this in your own tap, or using different software to help you package your app.

**Reply by pypy702** [reply]
Thanks for the reply.

> We can't guarantee that software built for one version of macOS will run on earlier or later versions as our continuous integration doesn't test for this. We do not have the resources to expand our testing to cover this use case.
>

If SDL2 doesn't work on later versions on macOS than targeted, then that would be an issue on their side.
Also, it's not guaranteeing anything. The Qt casks target macOS 10.13 for example. That's the minimum supported macOS version of Qt.
Likewise, the minimum macOS supported version of SDL2 is macOS 10.6, the version that they put into their script. And the script is their recommended way to build it.

**Reply by jonchang** [reply]
While that might be true, it'll still be something the Homebrew maintainers have to deal with and fix, since we are the ones packaging the software. Since we don't have a way to automatically test for these potential issues, we won't support this use case.

**Reply by pypy702** [reply]
I don't see what potential issues could pop up just by using the script to build it.
I myself don't need to support as low as 10.6. I am on macOS 11 myself, and there are many users on macOS 10.15 and macOS 10.14, which are officially supported versions of macOS by Homebrew.

**Reply by jonchang** [reply]
This is becoming unproductive. I've already explained why we can't do this. Not being able to see any issues with this change is not the same as being able use continuous integration to test for the same issues. Homebrew does not have enough maintainers to do the work of manually verifying that software built on one version of macOS will correctly function on other versions of macOS. Since you are so familiar with sdl2 it's probably best for you to either do this yourself in your own tap, or write your own build scripts to package your app in the way you want. The Homebrew maintainers can't assist you with this.
